### PR TITLE
1399 migrate back to top testing over to cypress

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
Cypress e2e, a11y and visual tests added, previous duplicate tests removed as per cypress migration.

## Related issue
#1399 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

**Review by ad9242 — CHANGES_REQUESTED** [review]
could we add a test\screenshot where the back-to-top has focus, just to capture that?

File: packages/react/src/component-tests/IcBackToTop/IcBackToTop.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+describe("IcBackToTop", () => {
+  beforeEach(() => {
+    cy.injectAxe();
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
+  it("should be hidden", () => {
```
Comment (GCHQ-Developer-299) [inline packages/react/src/component-tests/IcBackToTop/IcBackToTop.cy.tsx:20]: This is a bit pedantic, but I think this test should have a more specific description. "should be hidden when viewport is at the top of the page"

**Review by GCHQ-Developer-530 — COMMENTED** [review]
Screenshot 2024-01-22 at 12 28 49
The baseline screenshot taken seems to have a ghost back to top further up. Not sure why this might be happening, but could be worth adding a wait before compareSnapshot to see if that gets rid of it?

File: packages/react/src/component-tests/IcBackToTop/IcBackToTop.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  it("should hide and scroll page to top when clicked", () => {
+    mount(<BackToTop />);
+
+    cy.scrollTo("bottom").checkHydrated("ic-back-to-top").wait(500);
+    cy.get("ic-back-to-top").shadow().find("button").should(BE_VISIBLE).click().wait(500);
+    cy.get("ic-back-to-top").shadow().find("button").should(NOT_BE_VISIBLE);
+  });
+
+  it("should render and pass a11y tests", () => {
```
Comment (GCHQ-Developer-299) [inline packages/react/src/component-tests/IcBackToTop/IcBackToTop.cy.tsx:61]: You could remove this test and add the checkA11y line into the first 'should appear' test.
The IcSelect tests are good examples of integrating Visual Regression and A11y checks into existing tests

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**